x-pack/test/functional/apps/canvas/index.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { EsArchiver } from '@kbn/es-archiver';
+import { FtrProviderContext } from '../../ftr_provider_context';
+export default ({ loadTestFile, getService }: FtrProviderContext) => {
+  const security = getService('security');
+  const config = getService('config');
+  let esNode: EsArchiver;
+
+  describe('Canvas', function canvasAppTestSuite() {
+    describe('Canvas app', () => {
+      before(async () => {
+        // init data
+        const roles = [
+          'test_logstash_reader',
+          'global_canvas_all',
+          'global_discover_all',
+          'global_maps_all',
+          // TODO: Fix permission check, save and return button is disabled when dashboard is disabled
+          'global_dashboard_all',
+        ];
+        if (config.get('esTestCluster.ccs')) roles.push('ccs_remote_search');
```

Comment:
Could be just a nit, but I see `config.get('esTestCluster.ccs')` evaluated like **4** times.  
You could do it once and assign to a variable, unless you thing it's value is changing out from under you and you need to to re-evaluate it every time.

Reply:
Fixed.